src/foam/nanos/auth/twofactor/GoogleTOTPAuthService.js

```diff
@@ -65,19 +68,20 @@ foam.CLASS({
         user.setTwoFactorSecret(key);
         userDAO.put_(x, user);
 
-        if ( ! generateQrCode ) {
-          return key;
-        }
-
         try {
           EmailConfig service = (EmailConfig) x.get("emailConfig");
           String name = service == null ? "FOAM" : service.getDisplayName();
           String path = String.format("/%s:%s", name, user.getEmail());
           String query = String.format("secret=%s&issuer=%s&algorithm=%s", key, name, getAlgorithm());
           URI uri = new URI("otpauth", "totp", path, query, null);
-          return "data:image/svg+xml;charset=UTF-8," + QrCode.encodeText(uri.toASCIIString(), QrCode.Ecc.MEDIUM).toSvgString(0);
+
+          return new OTPKey.Builder(x)
+            .setKey(key)
+            .setQrCode("data:image/svg+xml;charset=UTF-8,"+
+              QrCode.encodeText(uri.toASCIIString(), QrCode.Ecc.MEDIUM).toSvgString(0))
+            .build();
         } catch ( Throwable t ) {
-          throw new RuntimeException(t);
+          throw new RuntimeException(t);	
```

Comment:
Remove trailing whitespace. Also it might be helpful to add some messaging to this exception so we can give a useful error message rather than just having the client receive "RuntimeException". Do we have an error type that the client knows how to interpret and display?